aws/resource_aws_glue_security_configuration.go

```diff
@@ -0,0 +1,342 @@
+		JobBookmarksEncryption: expandGlueJobBookmarksEncryption(m["job_bookmarks_encryption"].([]interface{})),
+		S3Encryption:           expandGlueS3Encryptions(m["s3_encryption"].([]interface{})),
+	}
+
+	return encryptionConfiguration
+}
+
+func expandGlueJobBookmarksEncryption(l []interface{}) *glue.JobBookmarksEncryption {
+	if len(l) == 0 || l[0] == nil {
+		return nil
+	}
+
+	m := l[0].(map[string]interface{})
+
+	jobBookmarksEncryption := &glue.JobBookmarksEncryption{
+		JobBookmarksEncryptionMode: aws.String(m["job_bookmarks_encryption_mode"].(string)),
+	}
+
+	if v, ok := m["kms_key_arn"]; ok {
+		jobBookmarksEncryption.KmsKeyArn = aws.String(v.(string))
+	}
+
+	return jobBookmarksEncryption
+}
+
+func expandGlueS3Encryptions(l []interface{}) []*glue.S3Encryption {
+	s3Encryptions := make([]*glue.S3Encryption, len(l))
+
+	for i, s3Encryption := range l {
+		if s3Encryption == nil {
+			continue
+		}
+		s3Encryptions[i] = expandGlueS3Encryption(s3Encryption.(map[string]interface{}))
+	}
+
+	return s3Encryptions
+}
+
+func expandGlueS3Encryption(m map[string]interface{}) *glue.S3Encryption {
+	s3Encryption := &glue.S3Encryption{
+		S3EncryptionMode: aws.String(m["s3_encryption_mode"].(string)),
+	}
+
+	if v, ok := m["kms_key_arn"]; ok {
+		s3Encryption.KmsKeyArn = aws.String(v.(string))
+	}
+
+	return s3Encryption
+}
+
+func flattenGlueCloudWatchEncryption(cloudwatchEncryption *glue.CloudWatchEncryption) []interface{} {
+	if cloudwatchEncryption == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"cloudwatch_encryption_mode": aws.StringValue(cloudwatchEncryption.CloudWatchEncryptionMode),
+		"kms_key_arn":                aws.StringValue(cloudwatchEncryption.KmsKeyArn),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenGlueEncryptionConfiguration(encryptionConfiguration *glue.EncryptionConfiguration) []interface{} {
+	if encryptionConfiguration == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"cloudwatch_encryption":    flattenGlueCloudWatchEncryption(encryptionConfiguration.CloudWatchEncryption),
+		"job_bookmarks_encryption": flattenGlueJobBookmarksEncryption(encryptionConfiguration.JobBookmarksEncryption),
+		"s3_encryption":            flattenGlueS3Encryptions(encryptionConfiguration.S3Encryption),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenGlueJobBookmarksEncryption(jobBookmarksEncryption *glue.JobBookmarksEncryption) []interface{} {
+	if jobBookmarksEncryption == nil {
+		return []interface{}{}
+	}
+
+	m := map[string]interface{}{
+		"job_bookmarks_encryption_mode": aws.StringValue(jobBookmarksEncryption.JobBookmarksEncryptionMode),
+		"kms_key_arn":                   aws.StringValue(jobBookmarksEncryption.KmsKeyArn),
+	}
+
+	return []interface{}{m}
+}
+
+func flattenGlueS3Encryptions(s3Encryptions []*glue.S3Encryption) []interface{} {
+	l := make([]interface{}, len(s3Encryptions))
```

Comment:
(as above) you may want to default this to 0 since we're skipping items below

Reply:
👍